dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/resource/ui/UISoftwareResource.java

```diff
@@ -72,64 +68,63 @@
   @GET
   @Produces(APPLICATION_JSON)
   public List<MavenArtifactVersion> getArtifacts() {
-    val results = softwareService.getMavenVersions("icgc-storage-client", properties);
+    val results = softwareService.getMavenVersions("icgc-storage-client");
     return results;
   }
 
   @Path("/icgc-get/versions")
   @GET
   @Produces(APPLICATION_JSON)
   public List<ArtifactFolder> getICGCGetArtifacts() {
-    val results = softwareService.getVersions("icgc-get", properties);
+    val results = softwareService.getVersions("icgc-get");
     return results;
   }
 
   @Path("/icgc-storage-client/latest")
   @GET
   @SneakyThrows
   public Response getLatest() {
-    URL redirect = new URL(softwareService.getLatestVersionUrl(properties));
+    URL redirect = new URL(softwareService.getLatestVersionUrl());
     return Response.seeOther(redirect.toURI()).build();
   }
 
   @Path("/icgc-get/latest/{os}")
   @GET
   @SneakyThrows
   public Response getICGCGetLatest(@PathParam("os") String os) {
-    URL redirect = new URL(softwareService.getLatestICGCGetVersionUrl(os, properties));
+    URL redirect = new URL(softwareService.getLatestICGCGetVersionUrl(os));
     return Response.seeOther(redirect.toURI()).build();
   }
 
-  
   @Path("/icgc-storage-client/{version}")
   @GET
   @SneakyThrows
   public Response getVersion(@PathParam("version") String version) {
-    URL redirect = new URL(softwareService.getVersionUrl(version, "icgc-storage-client", properties));
+    URL redirect = new URL(softwareService.getVersionUrl(version, "icgc-storage-client"));
     return Response.seeOther(redirect.toURI()).build();
   }
-  
+
   @Path("/icgc-get/{version}/{os}")
   @GET
   @SneakyThrows
   public Response getICGCGetLinuxVersion(@PathParam("version") String version, @PathParam("os") String os) {
-    URL redirect = new URL(softwareService.getICGCGetVersionUrl(version, os, properties));
+    URL redirect = new URL(softwareService.getICGCGetVersionUrl(version, os));
```

Comment:
Why do some service methods take a string, while others bake it into the name? I think it should be the latter. No need for the resource to know about the internal resolution details.